pkg/resourceinterpreter/subresourceinterpreter/subresource.go

```diff
@@ -0,0 +1,78 @@
+package subresourceinterpreter
+
+import (
+	"context"
+
+	autoscalingapi "k8s.io/api/autoscaling/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/scale"
+	"k8s.io/klog/v2"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+)
+
+// SubResourceInterpreter interpret resource with the related subResource.
+type SubResourceInterpreter struct {
+	restMapper  meta.RESTMapper
+	scaleClient scale.ScalesGetter
+}
+
+// NewSubResourceInterpreter return a new SubResourceInterpreter.
+func NewSubResourceInterpreter(restMapper meta.RESTMapper, scaleClient scale.ScalesGetter) *SubResourceInterpreter {
+	return &SubResourceInterpreter{
+		restMapper:  restMapper,
+		scaleClient: scaleClient,
+	}
+}
+
+// HookEnabled tells if any hook exist for specific resource type and operation type.
+func (i *SubResourceInterpreter) HookEnabled(object *unstructured.Unstructured, operation configv1alpha1.InterpreterOperation) bool {
+	switch operation {
+	case configv1alpha1.InterpreterOperationInterpretReplica:
+		return i.enabledInterpretReplica(object)
+	}
+
+	// other operations are not supported currently.
```

Comment:
`currently`? Do you have a plan to introduce other operations in the future?

Reply:
`ReviseReplica` will be added, refer to #2486.